guides/keras_cv/segment_anything_in_keras_cv.py

```diff
@@ -0,0 +1,500 @@
+possible segmentation masks.
+"""
+
+"""
+## Box Prompts
+
+Now, let's see how we can prompt the model using boxes. The box is specified using two
+points, the top-left corner and the bottom-right corner of the bounding box in xyxy
+format. Let's prompt the model using a bounding box around the left front tyre of the
+truck.
+"""
+
+# Let's specify the box
+input_box = np.array([[240, 340], [400, 500]])
+
+outputs = model.predict(
+    {"images": image[np.newaxis, ...], "boxes": input_box[np.newaxis, np.newaxis, ...]}
+)
+mask = inference_resizing(outputs["masks"][0][0][..., None], pad=False)[..., 0]
+mask = ops.convert_to_numpy(mask) > 0.0
+
+plt.figure(figsize=(10, 10))
+plt.imshow(ops.convert_to_numpy(image) / 255.0)
+show_mask(mask, plt.gca())
+show_box(input_box, plt.gca())
+plt.axis("off")
+plt.show()
+
+"""
+Boom! The model perfectly segments out the left front tyre in our bounding box.
+
+## Combining prompts
+
+To get the true potential of the model out, let's combine box and point prompts and see
+what the model does.
+"""
+
+# Let's specify the box
+input_box = np.array([[240, 340], [400, 500]])
+# Let's specify the point and mark it background
+input_point = np.array([[325, 425]])
+input_label = np.array([0])
+
+outputs = model.predict(
+    {
+        "images": image[np.newaxis, ...],
+        "points": input_point[np.newaxis, ...],
+        "labels": input_label[np.newaxis, ...],
+        "boxes": input_box[np.newaxis, np.newaxis, ...],
+    }
+)
+mask = inference_resizing(outputs["masks"][0][0][..., None], pad=False)[..., 0]
+mask = ops.convert_to_numpy(mask) > 0.0
+
+plt.figure(figsize=(10, 10))
+plt.imshow(ops.convert_to_numpy(image) / 255.0)
+show_mask(mask, plt.gca())
+show_box(input_box, plt.gca())
+show_points(input_point, input_label, plt.gca())
+plt.axis("off")
+plt.show()
+
+"""
+Voila! The model understood that the object we wanted to exclude from our mask was the
+rim of the tyre.
+
+## Text prompts
+
+Finally, let's see how text prompts can be used along with KerasCV's
+`SegmentAnythingModel`.
+
+For this demo, we will use the [offical grounding dino
+model](https://github.com/IDEA-Research/GroundingDINO). Grounding DINO is a model that
+takes as input a `(image, text)` pair and generates a bounding box around the object in
+the `image` described by the `text`. You can refer to the
+[paper](https://arxiv.org/abs/2303.05499) for more details on the implementation of the
+model.
+
+For this part of the demo, we will need to install the `groundingdino` package from
+source:
+
+```
+pip install -U git+https://github.com/IDEA-Research/GroundingDINO.git
+```
+
+Then, we can install the pretrained model's weights and config:
+"""
+
+"""shell
+wget -q https://github.com/IDEA-Research/GroundingDINO/releases/download/v0.1.0-alpha/groundingdino_swint_ogc.pth
+wget -q https://raw.githubusercontent.com/IDEA-Research/GroundingDINO/main/groundingdino/config/GroundingDINO_SwinT_OGC.py
```

Comment:
Pulling from `main` might break in the future, can you pull from the 0.1.0 release?

Reply:
Done.